src/dbus/server/dbus_thread_object.cpp

```diff
@@ -1522,6 +1524,22 @@ void DBusThreadObject::ActiveDatasetChangeHandler(const otOperationalDatasetTlvs
     SignalPropertyChanged(OTBR_DBUS_THREAD_INTERFACE, OTBR_DBUS_PROPERTY_ACTIVE_DATASET_TLVS, value);
 }
 
+void DBusThreadObject::LeaveNetworkHandler(DBusRequest &aRequest)
+{
+    mNcp->GetThreadHelper()->DetachGracefully([aRequest, this](otError error) mutable {
+        SuccessOrExit(error);
+        SuccessOrExit(error = otInstanceErasePersistentInfo(mNcp->GetThreadHelper()->GetInstance()));
+
+    exit:
+        aRequest.ReplyOtResult(error);
+        if (error == OT_ERROR_NONE)
```

Comment:
what if detach succeeded but erasing settings failed? should we exit in such case?

Reply:
We should not exit if there are errors, because the operation is not complete.

In this case, erasing fails only when state is not disabled. It's not possible here.